[Snackbar] Display stacked Snackbars (#31991)
mui/material-ui: docs/data/material/components/multiple-snackbars/multiple-snackbars.md

The `action` prop can be used to show content on the right side of the snackbar.
It can simply be a React Node or a callback function with a key parameter returning a React Node.

Use the `snackbars.closeSnackbar` method to dismiss a specific snackbar by providing the `key` from the `action` callback argument.

{{"demo": "SimpleSnackbars.js", "defaultCodeOpen": false}}

## Customization

Use the `content` prop to customize the individual snackbars.

{{"demo": "CustomizedSnackbars.js", "defaultCodeOpen": false}}

## Positioned snackbars

You can control the position of an individual snackbar by specifying the `anchorOrigin` prop.
For even greater convience, pass the desired `anchorOrigin` value to the `SnackbarsProvider` component to set the position of all snackbars used globally.

{{"demo": "PositionedSnackbars.js", "defaultCodeOpen": false}}

## Transition

Similarly to positioning, use the`TransitionComponent` prop in an individual snackbar if you want to control that specific instance transition effect or use it in the `SnackbarsProvider` component to change it globally.

{{"demo": "TransitionedSnackbars.js", "defaultCodeOpen": false}}

## Props priority

> For the mutual properties between `showSnackbar` options and `SnackbarsProvider` props, the options in `showSnackbar` method of `useSnackbars` hook for individual snackbars take more preference over props passed in `SnackbarsProvider` if both are supplied.

Reviewer: I've had a hard time trying to understand this paragraph and I think I haven't yet 😅 Is there a way we could try to simplify it?

Author: I agree it is confusing, how about this:
```suggestion
In the event, when there are same props supplied for both `showSnackbar` and `SnackbarsProvider`, the options in `showSnackbar` method will gain priority over props in `SnackbarsProvider`. 
```

---
product: material-ui
title: Multiple Snackbars React component
components: SnackbarsProvider
materialDesign: https://material.io/components/snackbars
waiAria: https://www.w3.org/TR/wai-aria-1.1/#alert
---

> # Multiple Snackbars

Reviewer 2: @samuelsycamore could you please review the content of this page?

Reviewer 2: Also, what should we do with:
1. https://mui.com/material-ui/react-snackbar/#consecutive-snackbars
2. https://deploy-preview-31991--material-ui.netlify.app/material-ui/react-snackbar/#notistack

Reviewer: I've done a bit of writing revision there but would love Sam's input as well regardless!

I've got a bit confused by the name in relation to what's being introduced. As far as I understood, there's a new `SnackbarsProvider` component and a `useSnackbars` hook. First question would be: wouldn't make more sense for the hook to be in Base? Second is: instead of placing the `SnackbarsProvider` in the lab, shouldn't we add it as a supporting component for the Snackbar?

That would allow having all the content inside the Snackbar, which seems to make a lot more sense to me. We'd avoid content and demo repetition while also connecting the narrative that notistack used to be what enabled multiple snackbars but now we have a "native" solution for that, "deprecating" the need to depend on notistack for that.

Author: `useSnackbars` is not a low level hook for unstyled Snackbar to be it in Base. It's a hook available for the applications where it can be used within their components to trigger a snackbar inside a `SnackbarsProvider` context.

While I agree that I will be easier to manage if it is moved within core's Snackbar for all the reasons you mentioned, it is supposed to be in lab since this is first version and we can would get more feedback from users regarding bugs, design, API design which is what lab is used for initially. 
Once it's robust, we move it to core which will make things much easier then (regarding repetitive demos etc.)

For now we can still show there is a replacement of notistack available in lab by having a link to this page. I have also added a link to multiple snackbars from Core's snackbar [here](https://deploy-preview-31991--material-ui.netlify.app/material-ui/react-snackbar/#main-content) in the `Frequency` section.  

What do you think?

Reviewer: Right, that makes sense. I changed the word slightly from _multiple_ to _stacked_ snackbars, which sounds clearer to me. Additionally, edited the documentation to showcase that notistack and now this component are both alternatives to stack snackbars, trying to make it clear that the latter is our native take on it.
